ipam/testutils_test.go

```diff
@@ -232,27 +246,29 @@ type TestGossipRouterClient struct {
 }
 
 func (grouter *TestGossipRouter) connect(sender router.PeerName, gossiper router.Gossiper) router.Gossip {
-	gossipChan := make(chan gossipMessage, 100)
+	gossipChan := make(chan interface{}, 100)
 
 	go func() {
 		gossipTimer := time.Tick(10 * time.Second)
 		for {
 			select {
-			case message := <-gossipChan:
-				if message.exitChan != nil {
-					message.exitChan <- true
+			case gossip := <-gossipChan:
+				if message, isExit := gossip.(exitMessage); isExit {
+					close(message.exitChan)
 					return
 				}
-
+				if message, isFlush := gossip.(flushMessage); isFlush {
+					close(message.flushChan)
+				}
```

Comment:
I reckon this would be easier to follow if the above were part of the type dispatch below. This does mean duplicating the loss conditional, but I think that is justifiable in this case.